homeassistant/components/cast/media_player.py

```diff
@@ -347,11 +401,85 @@ def new_connection_status(self, connection_status):
         if self._valid:
             self._cast_device.new_connection_status(connection_status)
 
+    @staticmethod
+    def added_to_multizone(group_uuid):
+        """Handle the cast added to a group."""
+        pass
+
+    def removed_from_multizone(self, group_uuid):
+        """Handle the cast removed from a group."""
+        if self._valid:
+            self._cast_device.multizone_new_media_status(group_uuid, None)
+            self._cast_device.multizone_new_cast_status(group_uuid, None)
+
+    def multizone_new_cast_status(self, group_uuid, cast_status):
+        """Handle reception of a new MediaStatus for a group."""
+        if self._valid:
+            self._cast_device.multizone_new_cast_status(
+                group_uuid, cast_status)
+
+    def multizone_new_media_status(self, group_uuid, media_status):
+        """Handle reception of a new MediaStatus for a group."""
+        if self._valid:
+            self._cast_device.multizone_new_media_status(
+                group_uuid, media_status)
+
+    def invalidate(self):
+        """Invalidate this status listener.
+
+        All following callbacks won't be forwarded.
+        """
+        # pylint: disable=protected-access
+        if self._cast_device._cast_info.is_audio_group:
+            self._mz_mgr.remove_multizone(self._uuid)
+        else:
+            self._mz_mgr.deregister_listener(self._uuid, self)
+        self._valid = False
+
+
+class DynamicGroupCastStatusListener:
+    """Helper class to handle pychromecast status callbacks.
+
+    Necessary because a CastDevice entity can create a new socket client
+    and therefore callbacks from multiple chromecast connections can
+    potentially arrive. This class allows invalidating past chromecast objects.
+    """
+
+    def __init__(self, cast_device, chromecast, mz_mgr):
+        """Initialize the status listener."""
+        self._cast_device = cast_device
+        self._uuid = chromecast.uuid
```

Comment:
The parameter `chromecast` is a bit ambiguous here. Is this the info ?

Reply:
It's an actual pychromecast.Chromecast object.